pywren_ibm_cloud/libs/docker/clients/ibm_vpc/ibm_vpc.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import requests
+
+from pywren_ibm_cloud.libs.ibm_utils import IBMIAMAPIKeyManager
+
+logger = logging.getLogger(__name__)
+
+
+class IBMVPCInstanceClient:
+
+    def __init__(self, ibm_vpc_config, insecure=False, user_agent=None):
+        self.config = ibm_vpc_config
+
+        self.session = requests.session()
+        if insecure:
+            self.session.verify = False
+
+        iam_api_key = self.config.get('iam_api_key')
+        token = self.config.get('token', None)
+        token_expiry_time = self.config.get('token_expiry_time', None)
+        self.ibm_iam_api_key_manager = IBMIAMAPIKeyManager('docker', iam_api_key, token, token_expiry_time)
+
+        headers = {'content-type': 'application/json'}
+        if user_agent:
+            default_user_agent = self.session.headers['User-Agent']
+            headers['User-Agent'] = default_user_agent + ' {}'.format(user_agent)
+        self.session.headers.update(headers)
+
+        adapter = requests.adapters.HTTPAdapter()
+        self.session.mount('https://', adapter)
+
+    def _authorize_session(self):
+        self.config['token'], self.config['token_expiry_time'] = self.ibm_iam_api_key_manager.get_token()
```

Comment:
@omerb01 is this generate new token every time this method is called? I think it is... can you check it? you don't need Gen2 for this

Reply:
@gilv nope, it will generate a new token only if it is expired :)